lib/utils/getAllowance/index.ts

```diff
@@ -0,0 +1,44 @@
+import { BigNumber, ethers } from "ethers";
+import { addresses, allowancesContracts } from "../../constants";
+import {
+  AllowancesToken,
+  AllowancesSpender,
+  Allowances,
+} from "../../types/allowances";
+
+export const isSpenderInAddresses = (spender: string): boolean => {
+  const keys = Object.keys(
+    addresses.mainnet
+  ) as (keyof typeof addresses["mainnet"])[];
+  return keys.includes(spender as keyof typeof addresses["mainnet"]);
+};
+
+export const getAllowance = async (params: {
```

Comment:
Yeah, I came up with this pattern because two many spenders with the same tokens!
How do you know in the end which BigNumber belonged to which token from which spender?

You pass in spender: 
"wsklima" | "retirementAggregator" | "staking_helper" | "staking" | "pklima_exercise";
for tokens:
"bct" | "mco2" | "nct" | "ubo" | "nbo" | "klima" | "pklima" | "sklima" | "wsklima" | "usdc"

And after that you need to know which BigNumber belonged to which token from which spender.

Let me know if you see an easier solution.

Reply:
So for that nested `forEach` here https://github.com/KlimaDAO/klimadao/pull/495/files#diff-254c9445c732e1b66b745e7db313501080a4ab658c580ed8584ed29502ecd194R111

we can track the index like `spender[spenderIndex]` and `token[tokenIndex]`

roughly what i have in mind:
```ts
// outer loop
spenders.reduce(state, value, spenderIndex => {

  // create array of strings
  const allowanceValues = tokenNames.map(getAllowance);

  // assign to inner object
  const spenderAllowances = allowanceValues.reduce((prev, value, i) => {
    const token = tokenNames[i]; // use index
    return { ...prev, [token]: value };
  });
  
  // assign the inner object to the final outer obj
  const spender = spenders[spenderIndex];
  return { ...prev, state[spender]: spenderAllowances }
});
```

Maybe its just my style but its more readable than a deep merge to me and preferable because it keeps the getAllowance util simple and predictable returning just a string!

Reply:
That´s the thing, 
`getAllowance` is an async function
So you need to wrap the whole reduce function in an async await in order to not loose the information which token allowance value belongs to which spender. If you want to nest this information still.

Reply:
ok if you want to take a stab at nested async loops or promise arrays or something please go for it otherwise I guess I can live with this returning an object :)

Reply:
I´d leave it as is right now.

Let´s improve in a later PR